apps/backend/src/app/modules/admin/admin.controller.ts

```diff
@@ -63,9 +67,36 @@ export class AdminController {
     private readonly mapsService: MapsService,
     private readonly mapReviewService: MapReviewService,
     private readonly usersService: UsersService,
-    private readonly adminActivityService: AdminActivityService
+    private readonly adminActivityService: AdminActivityService,
+    private readonly killswitchService: KillswitchService
   ) {}
 
+  @Patch('/killswitch')
+  @HttpCode(HttpStatus.NO_CONTENT)
+  @ApiOperation({ summary: 'Update active killswitches' })
+  @ApiNoContentResponse({
+    description: 'The switches were updated successfully'
+  })
+  @ApiBadRequestResponse({ description: 'Invalid switches type' })
+  @ApiBody({
+    type: Object,
+    description: 'Object of type Record<KillswitchType, boolean>',
+    required: true
+  })
+  updateKillSwitch(@Body() switches: Killswitches): void {
+    this.killswitchService.updateKillswitches(switches).then((r) => r);
+  }
```

Comment:
`.then((r) => r)` have no effect whatever besides make ESLint think you're not ignoring a promise. `.then` behaves kind like `.map` on arrays, transforming whatever the promise returns after it resolves (if you want to be nerdy I *think* it's a monad). `.then(r => r)` says, "when this promise resolves, map `r` to `r` - has no absolutely no affect.

In this case you just want to `await` the Promise and not return it. That ensures the work being done inside the promise actually completes before we return a response, but we don't actually return whatever the promise resolves out to. In your case you're not actually awaiting anything, this endpoint will return as soon as `updateKillswitches` returns a promise, before the ongoing async work in that method has completed.

Reply:
Updated in [91fe759](https://github.com/momentum-mod/website/commit/91fe759399519f03c2b85ca32f85968d4e0d5ea7)